Fail new builds that can't start build pod because it already exists

mmilata opened the pull request:
Consider following scenario:

- build is started by user
- build controller creates build pod
- build is deleted by user
- build pod is killed but remains Terminating during grace period
- build with same name as in first step is started by user

Previously the new build stayed in New phase until the old build pod finished and then went into Failed after the pod terminated. This commit causes the new build to go into Error phase when it first tries creating build pod and fails because pod with the same name exists.

Slightly improves bug 1300949 (the described behavior is still very similar).

bparees commented on pkg/build/controller/controller.go:
```diff
@@ -201,8 +201,11 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 			glog.V(4).Infof("Build pod already existed: %#v", podSpec)
+			build.Status.Phase = buildapi.BuildPhaseError
+			build.Status.Reason = buildapi.StatusReasonBuildPodExists
+			build.Status.Message = err.Error()
```
We need to be checking if the pod is older than the build.  if the pod is newer than the build, we should continue to no-op this (it indicates something else managed to create the pod for us).

mmilata replied:
Ah, sorry, totally forgot about this - updated.

mmilata commented:
[test][testextended][extended:core(builds)]

bparees commented:
lgtm [merge]

mmilata commented:
Rebased, also updated the code to be in line with #10817 - the message field is set to predefined message instead of propagating the AlreadyExists error.

bparees commented:
[merge]

bparees commented:
extended test failure is a known issue.
